app/forms/candidate_interface/degree_wizard.rb

```diff
@@ -0,0 +1,447 @@
+          subject: subject,
+          subject_hesa_code: hesa_subject_code(subject),
+          grade: grade_attributes,
+          grade_hesa_code: hesa_grade_code(grade_attributes),
+          predicted_grade: predicted_grade,
+          start_year: start_year,
+          award_year: award_year,
+        }
+      else
+        {
+          application_form_id: application_form_id,
+          level: 'degree',
+          international: true,
+          institution_country: country,
+          qualification_type: international_type,
+          institution_name: university,
+          subject: subject,
+          predicted_grade: predicted_grade,
+          grade: other_grade || map_value_for_no_submitted_international_grade(grade),
+          start_year: start_year,
+          award_year: award_year,
+          enic_reference: enic_reference,
+          comparable_uk_degree: comparable_uk_degree,
+        }
+      end
+    end
+
+    def sanitize_attrs(attrs)
+      sanitize_uk_or_non_uk(attrs)
+      sanitize_country(attrs)
+      sanitize_type(attrs)
+      sanitize_degree_level(attrs)
+      sanitize_grade(attrs)
+      sanitize_enic(attrs)
+      attrs
+    end
+
+    def international?
+      uk_or_non_uk == 'non_uk'
+    end
+
+    def hesa_institution_code(institution_name)
+      Hesa::Institution.find_by_name(institution_name)&.hesa_code
+    end
+
+    def hesa_type_code(type_description)
+      Hesa::DegreeType.find_by_name(type_description)&.hesa_code
+    end
+
+    def hesa_subject_code(subject)
+      Hesa::Subject.find_by_name(subject)&.hesa_code
+    end
+
+    def hesa_grade_code(grade)
+      Hesa::Grade.find_by_description(grade)&.hesa_code
+    end
+
+    def qualification_type_attributes
+      return 'Level 6 Diploma' if degree_level == 'Level 6 Diploma'
+
+      equivalent_level || other_type || type
+    end
+
+    def grade_attributes
+      return other_grade if grade == 'Other'
+
+      grade || other_grade
+    end
+
+    def predicted_grade
+      completed == 'No'
+    end
+
+    def completed?
+      completed == 'Yes'
+    end
+
+    def uk?
+      uk_or_non_uk == 'uk'
+    end
+
+    def other_qualification?
+      degree_level == 'Another qualification equivalent to a degree'
+    end
+
+    def level_options?
+      ['Level 6 Diploma', 'Another qualification equivalent to a degree'].include?(degree_level)
+    end
+
+    def other_type_selected
+      type == "Another #{degree_level.split.first.downcase} degree type"
+    end
+
+    def other_type_filled_out
+      other_type.present?
+    end
+
+    def grade_choices
+      grade == 'Other' || grade == 'Yes'
+    end
+
+    def subjects
+      @subjects ||= Hesa::Subject.names.sort
+    end
+
+    def institutions
+      @institutions ||= Hesa::Institution.names.sort
+    end
+
+    def dynamic_type(degree_level)
+      return if degree_level.nil?
+      return 'doctorate' if degree_level == 'Doctorate (PhD)'
```

Comment:
Should the two hard-coded strings here be constants?

Reply:
Yep, thanks, i've added the constants